### PR TITLE
noti: add module

### PR DESCRIPTION
**Pull request by marsam**
Add module for https://github.com/variadico/noti

### CODE REVIEW

File: modules/programs/noti.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+in
+
+{
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.noti = {
+    enable = mkEnableOption "Noti";
+
+    settings = mkOption {
+      type = types.attrs;
```
Comment (rycee) [inline modules/programs/noti.nix]: Had a quick look at [noti.md](https://github.com/variadico/noti/blob/master/docs/noti.md) and from what I can tell all options expect a string value so here you could use `types.attrsOf types.str`.

File: modules/programs/noti.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      default = {};
+      description = ''
+        Configuration written to
+        <filename>~/.config/noti/noti.yaml</filename>.
+
```
Comment (rycee) [inline modules/programs/noti.nix]: Need
```xml
</para><para>
```
to indicate paragraph separation.

File: modules/programs/noti.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        See
+        <citerefentry>
+          <refentrytitle>noti.yaml</refentrytitle>
+          <manvolnum>5</manvolnum>
+        </citerefentry>.
+        for the full list of options.
+      '';
+      example = {
```
Comment (rycee) [inline modules/programs/noti.nix]: I think you'll want `literalExample` here to make the example show up a bit nicer in the documentation. I.e., something like
```nix
example = literalExample ''
  {
    say = {
      voice = "Alex";
    };
    slack = {
      token = "1234567890abcdefg";
      channel = "@jaime";
      username = "noti";
    };
  };
'';
```
Alternatively just shorten it to something like
```nix
example = { say = { voice = "Alex"; }; };
```

File: modules/programs/noti.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        slack = {
+          token = "1234567890abcdefg";
+          channel = "@jaime";
+          username = "noti";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (
```
Comment (rycee) [inline modules/programs/noti.nix]: Alternatively, the shorter
```nix
config = mkIf cfg.enable {
  home.packages = [ pkgs.noti ];

  xdg.configFile."noti/noti.yaml" = mkIf (cfg.settings != {}) {
    text = generators.toYAML {} cfg.settings;
  };
};
```
should work.

**Comment by marsam** [comment]
@rycee thanks, I've made the changes you've suggested

**Comment by rycee** [comment]
Great, it looks good! I've rebased to master in 3f34bf4465b367c6eb4bc109349e75638e64ae52. Thanks for the contribution 👍

**Pull request closed without merging**